### DISCUSSION TITLE
How to switch to production mode on Azure?

### DISCUSSION THREAD

**Discussion by Zumpel96** [keycloak/keycloak#17575] [Q&A] [answered]
Hello!

I have a Azure App Service (Docker Container), which pulls the image according to the [documentation ](https://www.keycloak.org/getting-started/getting-started-docker). From what I have seen, to switch from dev (`start-dev`) to production (``start`), I need 2 things:

- Adding SSL
- Adding a sophisticated database (that's not the problem)

Now the thing is the following: I have my domain bound to the Azure App Service and it is already configured to only serve over HTTPS. Therefore I used the proxy setting (`KC_PROXY`) and set it to `passthrough`. This works like a charm. 

Since I use a proxy already, I therefore find it kinda obsolete and redundant to add the certificate - I have to add: I am not a security pro. Additionally, since I configured everything with environment variables, I don't know what the appropriate next steps are to make it run in production mode directly in Azure.

Help is highly appreciated, since this seems like a super straightforward thing and is not documented anywhere (at least I wasn't able to find it).

**Comment by maveonair** [comment, marked as answer]
If you care about HTTPS being terminated at the Azure App Service level, then you can set `KC_RPOXY` to `edge` instead of `passthrough`. With this setting, you do not need to provide a SSL certificate to run Keycloak. Instead, you tell Keycloak that you will communicate over HTTP between your proxy (Azure App Service) and Keycloak itself.  See the documentation for more details: https://www.keycloak.org/server/reverseproxy. 

I have been using Keycloak in this configuration with App Service for many months with no problems.

**Reply by Zumpel96** [reply]
Ah nice, thank you. I had a misunderstanding here and thought I might run into problems, when the DB and the App Service are not run within the same network. Good to know, thanks!

**Reply by maveonair** [reply]
Well, I would recommend that you put the DB and your Azure App Service in the same VNET and use a Private Endpoint to connect from the App Service to the DB.
